pkg/builder/statefulset_builder.go

```diff
@@ -157,12 +158,13 @@ func buildStsVolumeClaimTemplates(mariadb *mariadbv1alpha1.MariaDB) []corev1.Per
 }
 
 func buildStsServiceAccountName(mariadb *mariadbv1alpha1.MariaDB) (autoMount *bool, serviceAccount string) {
-	if mariadb.Galera().Enabled {
-		mount := false
-		autoMount = &mount
-		serviceAccount = mariadb.Name
+	if mariadb.Spec.ServiceAccountName != nil {
+		if mariadb.Spec.Galera.Enabled {
+			return ptr.To(false), *mariadb.Spec.ServiceAccountName
+		}
+		return ptr.To(true), *mariadb.Spec.ServiceAccountName
```

Comment:
Do we need the automount = true? The serviceaccount token was mounted when the default SA was used, is it used?

Reply:
We can set it to false in the non Galera clusters, as we are not currently using it. For Galera, we are explicitly mounting it here:
https://github.com/mariadb-operator/mariadb-operator/blob/main/pkg/builder/statefulset_builder.go#L197

Juding by https://github.com/mariadb-operator/mariadb-operator/issues/244 and https://github.com/mariadb-operator/mariadb-operator/issues/333, this issue is about be compliant with security policies, which in general are against the default `ServiceAccount`.